persistence/src/main/java/com/epam/esm/dao/impl/CertificateDAO.java

```diff
@@ -0,0 +1,183 @@
+  }
+
+  /**
+   * Find all tags method method
+   *
+   * @return {@link java.util.Collection} of certificates
+   */
+  @Override
+  public Collection<Certificate> findAll() {
+    return jdbcTemplate.query(SQL_FIND_ALL, new CertificateMapper());
+  }
+
+  /**
+   * Find all certificates by tag id method
+   *
+   * @param tagId id of the tag for find
+   * @return {@link java.util.Collection} of certificates
+   */
+  @Override
+  public Collection<Certificate> findAll(BigInteger tagId) {
+    return jdbcTemplate.query(SQL_FIND_ALL_BY_TAG_ID, new CertificateMapper(), tagId.longValue());
+  }
+
+  /**
+   * Find all certificates by params
+   *
+   * @param tagName name of the tag
+   * @param name part of name of the certificate
+   * @param description part of description of the certificate
+   * @param sortName sort by name type(asc, desc)
+   * @param sortDate sort by date type(asc, desc)
+   * @return {@link java.util.Collection} of certificates by params
+   * @see com.epam.esm.search.SearchQueryBuilder
+   */
+  public Collection<Certificate> findAll(
+      String tagName, String name, String description, String sortName, String sortDate) {
+    final String searchQuery = getSearchQuery(tagName, name, description, sortName, sortDate);
+    return jdbcTemplate.query(searchQuery, new CertificateMapper());
+  }
+
+  /**
+   * Find certificates by tag name method
+   *
+   * @param tagName name of tag
+   * @return {@link java.util.Collection} of certificates
+   */
+  public Collection<Certificate> findAll(String tagName) {
+    return jdbcTemplate.query(SQL_FIND_ALL_BY_TAG_NAME, new CertificateMapper(), tagName);
+  }
+
+  /**
+   * Find certificate by id method
+   *
+   * @param id id of the certificate
+   * @return certificate or null if certificate not found
+   */
+  @Override
+  public Certificate find(BigInteger id) {
+    return jdbcTemplate.query(SQL_FIND_BY_ID, new CertificateMapper(), id.longValue()).stream()
+        .findAny()
+        .orElse(null);
+  }
+
+  /**
+   * Add certificate in database method
+   *
+   * @param certificate certificate entity
+   * @return count of added rows
+   */
+  @Override
+  public int add(Certificate certificate) {
+    return jdbcTemplate.update(
+        SQL_ADD,
+        certificate.getName(),
+        certificate.getDescription(),
+        certificate.getPrice(),
+        certificate.getDuration(),
+        certificate.getCreateDate(),
+        certificate.getLastUpdateDate());
+  }
+
+  /**
+   * Update certificate by id method
+   *
+   * @param id id of the certificate to update
+   * @param certificate certificate to update
+   * @return count of added rows
+   */
+  @Override
+  public int update(BigInteger id, Certificate certificate) {
+    return jdbcTemplate.update(
+        SQL_UPDATE,
+        certificate.getName(),
+        certificate.getDescription(),
+        certificate.getPrice(),
+        certificate.getDuration(),
+        certificate.getCreateDate(),
```

Comment:
@belikandrey I think we shouldn't give capability to update create date. Create date should be add once and never change